code-challenge-android-kotlin/app/src/main/java/com/arctouch/codechallenge/ui/HomeActivity.kt

```diff
@@ -0,0 +1,105 @@
+package com.arctouch.codechallenge.ui
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.view.View
+import com.arctouch.codechallenge.viewmodel.HomeViewModel
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.widget.AbsListView
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.data.Cache
+import com.arctouch.codechallenge.home.HomeAdapter
+import com.arctouch.codechallenge.model.Genre
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.webservice.MoviesWS
+import kotlinx.android.synthetic.main.home_activity.*
+
+class HomeActivity : AppCompatActivity() {
+    private var isScrolling = false
```

Comment:
From **line 21 to 26** you added some variables inside the UI. According to MVVM (and also MVP) its not View's concern to handle logics except UI logics.

It should be better to have those variables declared inside the ViewModel and handle them from there.

Reply:
Movie arraylist variable and pageNumber should be deleted as the MovieWS would be moved to the ViewModel (or repository as you mentioned). About the other variables, they store how the scrollListener is "executing". Should they be moved elsewhere too?

Reply:
Actually it should be inside an implementation of the ScrollListener. on that way, it should be better to concentrate the logic there so the View (Activity) must no nothing about any scrolling logic. Its only concern should be observing model changes (through ViewModel) stated by the MVVM pattern. As advice, just check this post regarding MVVM best practices and anti-pattern: https://medium.com/androiddevelopers/viewmodels-and-livedata-patterns-antipatterns-21efaef74a54